### PR TITLE
Replaces SpanStoreValidator with SpanStoreSpec

### PR DESCRIPTION
**Pull request by codefromthecrypt**
SpanStoreValidator was in src/main and included a bespoke unit test
framework. This is harder to understand than normal unit tests, clutters
the main source tree, and makes debugging test failures custom.

This introduces a similar class, SpanStoreSpec, which uses normal
assertions and supports JUnit annotations. This means that you can use 
IDE support such as rerunning a specific test, override or ignore test 
methods, or even use test rules to start your backends. Finally, this
gets the test code out of the main source tree.

### CODE REVIEW

**Comment by adriancole** [comment]
Don't merge, yet. I need to change this to skip the two failing cassie tests.

**Comment by codefromthecrypt** [comment]
ok PTAL

File: zipkin-anormdb/src/main/scala/com/twitter/zipkin/storage/anormdb/AnormSpanStore.scala [hunk 1]
```diff
@@ -116,7 +116,7 @@ class AnormSpanStore(
               ("value" -> Util.getArrayFromBuffer(b.value)),
               ("annotation_type_value" -> b.annotationType.value),
               ("ipv4" -> b.host.map(_.ipv4)),
-              ("port" -> b.host.map(_.ipv4)))
+              ("port" -> b.host.map(_.port)))
```
Comment (codefromthecrypt) [inline zipkin-anormdb/src/main/scala/com/twitter/zipkin/storage/anormdb/AnormSpanStore.scala:119]: noticed this bug lying around..

Comment (jamescway) [inline zipkin-anormdb/src/main/scala/com/twitter/zipkin/storage/anormdb/AnormSpanStore.scala:119]: hopefully its not a poorly documented hack O_o

**Comment by jamescway** [comment]
Interesting that you uncovered the testing framework.  I restarted the travis job, but there was a problem with the redis span store spec.  Not sure if this is an intermittent failure or not.

**Comment by codefromthecrypt** [comment]
yeah, well redis didn't actually run the tests prior (only did clear() actually!), so there may be some flakes to squash over time. beats zero coverage, though!

**Comment by codefromthecrypt** [comment]
ok build's green. I will help with flakes or disabling things that flake on certain backends.

**Pull request merged**